simulators/main/simulators.py

```diff
@@ -61,27 +62,54 @@ def __create_task(self, definition):
     def tick(self):
         if not self.machine.enabled:
             return
-        if self.machine.should_tick():
+        if self.should_tick():
             for task in self.machine.tasks:
-                try:
-                    if self.machine.first_time:
-                        # set the next_run time to always let the measurement generation to run in the first time
-                        task.next_run = datetime.timestamp(datetime.utcnow()) + 1
-                        self.machine.first_time = False
-                except:
-                    pass
+                self.is_first_time(task)
                 if task:
                     task.tick()
 
+    def should_tick(self):
+        if not self.is_in_productionTime():
+            # If machine is out of production time continuously, the log won't be generated
+            if not self.machine.out_of_production_time_logged:
+                self.log_not_in_shift()
+                self.machine.out_of_production_time_logged = True
+            return False
+        else:
+            self.machine.out_of_production_time_logged = False
+            return True
+
+    def is_first_time(self, task):
+        with contextlib.suppress(Exception):
+            if self.machine.first_time:
+                # set the next_run time to always let the measurement generation to run in the first time
+                task.next_run = datetime.timestamp(datetime.utcnow()) + 1
+                self.machine.first_time = False
+
+    def is_in_productionTime(self):
+        # if there are no shiftplans for a device, it should not be affected by production-time
+        has_no_shiftplan = True
+        for shiftplan in self.machine.shiftplans:
+            if shiftplan.locationId == self.machine.locationId:
+                has_no_shiftplan = False
```

Comment:
I think we dont need that logic anymore; the "/status" endpoint will also handle locations without a shift definition properly